RoslynPluginGenerator/AnalyzerPluginGenerator.cs

```diff
@@ -66,24 +84,16 @@ public bool Generate(NuGetReference analyzeRef, string sqaleFilePath)
 
                 string rulesFilePath = Path.Combine(outputDirectory, "rules.xml");
 
+                // TODO: we shouldn't try to work out where the content files have been installed by NuGet.
+                // Instead, we should use the methods on IPackage to locate the assemblies e.g. Package.GetFiles()
                 string packageDirectory = Path.Combine(nuGetDirectory, package.Id + "." + package.Version.ToString());
                 Debug.Assert(Directory.Exists(packageDirectory), "Expected package directory does not exist: {0}", packageDirectory);
+
                 bool success = TryGenerateRulesFile(packageDirectory, nuGetDirectory, rulesFilePath);
 
                 if (success)
                 {
-                    this.logger.LogInfo(UIResources.APG_GeneratingPlugin);
-
-                    string fullJarPath = Path.Combine(Directory.GetCurrentDirectory(), 
-                        analyzeRef.PackageId + "-plugin." + pluginDefn.Version + ".jar");
-
-                    PluginBuilder builder = new PluginBuilder(logger);
-                    RulesPluginBuilder.ConfigureBuilder(builder, pluginDefn, language, rulesFilePath, sqaleFilePath);
-
-                    builder.SetJarFilePath(fullJarPath);
-                    builder.Build();
-                    
-                    this.logger.LogInfo(UIResources.APG_PluginGenerated, fullJarPath);
```

Comment:
Extracted to a new method "BuildPlugin". The only change in the new method is the addition of the new line 195 i.e. the call to "AddRoslynMetadata".